### PR TITLE
Upgraded go version in pipeline

### PR DESCRIPTION
**Pull request by sssash18**
**What this PR does / why we need it**:
This PR updates the go version in pipeline.
**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```other operator
NONE
```

### CODE REVIEW

**Review by rishabh-11 — CHANGES_REQUESTED** [review]
one small change needed

**Review by elankath — APPROVED** [review]

File: .ci/pipeline_definitions [hunk 1]
```diff
@@ -36,9 +36,9 @@ autoscaler:
             dockerfile: './cluster-autoscaler/Dockerfile'
     steps:
       test:
-        image: 'golang:1.20.5'
+        image: 'golang:1.21.6'
```
Comment (rishabh-11) [inline .ci/pipeline_definitions]: Can you change it to `1.22.2` instead of `1.21.6`? Same in other places

**Review by rishabh-11 — CHANGES_REQUESTED** [review]

File: .ci/pipeline_definitions (hunk 1 above)
Comment (sssash18) [reply .ci/pipeline_definitions]: the go version in CA 1.29 is still 1.21, so not sure if this will work out.

Comment (rishabh-11) [reply .ci/pipeline_definitions]: It should work. The `go` version cannot be less than the required one. A higher version should be fine.

Comment (sssash18) [reply .ci/pipeline_definitions]: done

**Review by rishabh-11 — APPROVED** [review]
/lgtm

**Pull request merged**